tests/search/vespa/test_vespa_concept_filters.py

```diff
@@ -0,0 +1,86 @@
+from unittest.mock import patch
+
+import pytest
+
+from app.api.api_v1.routers import search
+from app.api.api_v1.schemas.search import SearchResponse
+from tests.search.vespa.setup_search_tests import (
+    _make_search_request,
+    _populate_db_families,
+)
+
+
+@pytest.mark.search
+@patch("app.api.api_v1.routers.search.verify_any_corpora_ids_in_db", return_value=True)
+@pytest.mark.parametrize(
+    "label,query,concept_filters",
+    [
+        ("search", "the", [{"name": "name", "value": "environment"}]),
+        (
+            "search",
+            "the",
+            [
+                {"name": "parent_concept_ids_flat", "value": "Q0,"},
+                {"name": "id", "value": "0"},
+            ],
+        ),
+        (
+            "browse",
+            "",
+            [
+                {"name": "model", "value": "sectors_1"},
+                {"name": "id", "value": "0"},
+            ],
+        ),
+    ],
+)
+def test_concept_filters(
+    mock_corpora_exist_in_db,
+    label,
+    query,
+    concept_filters,
+    test_vespa,
+    data_db,
+    monkeypatch,
+    data_client,
+    valid_token,
+):
+    monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
+
+    _populate_db_families(data_db, deterministic_metadata=True)
+
+    response = _make_search_request(
+        data_client,
+        valid_token,
+        {
+            "query_string": query,
+            "concept_filters": concept_filters,
+        },
+        expected_status_code=200,
+    )
+    response: SearchResponse = SearchResponse.model_validate(response)
+
+    assert len(response.families) > 0
+    for concept_filter in concept_filters:
```

Comment:
I know this is rather nested but it is a nested data structure and more readable than anything else I was thinking of refactoring to.

Reply:
Haha, makes sense. Fun times!

Reply:
You could use snapshot testing 😉 I'm happy to try that out quickly after this PR!

Reply:
With Marks help, I've got the snapshots generated!